bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -423,6 +414,25 @@ public static ResourcesPlugin getPlugin() {
 	 *         class.
 	 */
 	public static IWorkspace getWorkspace() {
+		ResourcesPlugin resourcesPlugin = plugin;
```

Comment:
I count ~1400 uses of this method in my SDK workspace and ~2600 in my product workspace.
Original code simply did a null check and return. No side effects, not much to worry about.

New code is overwhelmingly complex (for something that is supposed to return a value that is computed once) and adds calls to method that is not recommended to be called on activation (javadoc: `It is strongly recommended that waitForService is not used during the calling of the BundleActivator methods. BundleActivator methods are expected to complete in a short period of time.`). I see that this method **is** called in activators.

By all means, this simple getter shouldn't do any blocking operations or cause other side effects through the framework calls behind. The probability that one of the ~1000 callers will have a side effect (even if this a performance drop) is just too high.

Reply:
> So any of the consumers who is scared about the complexity here could simply migrate to consuming the service here

I'm scared because from a simple getter we get a complex code. This method is used a lot everywhere, since day one, and surely no one will migrate anything here to some complex services pattern.

Reply:
If you prefer to simply fail I can adjust, beside that a service pattern is nothing "complex", what makes it complex is that most Eclipse plugins don't use anything here but rely on static method calls instead.

Eg I recently refactored [CheckMissingNaturesListener](https://github.com/eclipse-platform/eclipse.platform.resources/blob/master/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CheckMissingNaturesListener.java) so I don't see it becomes more complex to have
```
@Reference
private IWorkspace workspace;
```
versus
```
ResourcesPlugin.getWorkspace()
```

beside at the latter I have to be scared to "unintentionally" activate the Resource plugin and run into all the trouble with Workspace (not yet) set.

Reply:
With https://github.com/eclipse-platform/eclipse.platform.resources/pull/87 I don't see any more calls to getWorkspace while openening, so once this is merged this method could be simplified again.